### PR TITLE
LIME-187: * fix routes and user refreshing

### LINKED ISSUE #187: BUG: Error messages "Token is required" are displayed when unauthorized user tries to reach Sign in or Sign up page
### Describe your issue

When unauthorized user is redirected from Home to Sign in page, two error messages "Token is required" are displayed to the user. When trying to reach Sign In and Sign Up through the URL, two error messages "Token is required" are displayed to the user. 401 status code is received for users/current request

### Steps to reproduce

1. Enter URL of the home page http://lime.eu-north-1.elasticbeanstalk.com/ in browser as an unauthorized user.
2. Check if Sign in page, where user is redirected, is displayed correctly.
3. Enter URL http://lime.eu-north-1.elasticbeanstalk.com/up in browser.
4. Check if Sign up page is displayed correctly.

### What was the actual result?

Sign in and Sign up pages are displayed with two error messages "Token is required". 401 status code is received for users/current request

### What was the expected result?

Sign In and Sign Up pages can be reached without messages of token required

### Put here any screenshots or videos (optional)

image
image

### What OS are you seeing the problem on?

Windows

### What browsers are you seeing the problem on?

Chrome

### Severity

Major (S3)

### PR DESCRIPTION
**Pull request by ivanbilobrov07**

### CODE REVIEW

File: frontend/src/bundles/auth/auth-api.ts [hunk 1]
```diff
@@ -44,6 +44,14 @@ class AuthApi extends BaseHttpApi {
...
         return await response.json<AuthResponseDto>();
     }
+
+    public async logout(): Promise<void> {
+        await this.load(this.getFullEndpoint(AuthApiPath.LOGOUT, {}), {
+            method: 'GET',
+            hasAuth: true,
+            contentType: ContentType.JSON,
+        });
+    }
```
Comment (dimapopovych) [inline frontend/src/bundles/auth/auth-api.ts:54]: why do we need this request?

File: frontend/src/framework/store/middlewares/error-middleware.ts [hunk 2]
```diff
@@ -1,14 +1,27 @@
...
+    dispatch: typeof store.instance.dispatch;
+}) => {
     return (next) => {
-        return (action) => {
+        return async (action) => {
             if (isRejected(action)) {
                 const { message = 'Something went wrong' } = action.error;
 
-                notificationManager.error(message);
+                if (message === UserValidationMessage.TOKEN_INVALID) {
```
Comment (dimapopovych) [inline frontend/src/framework/store/middlewares/error-middleware.ts]: I would either check status code or add separate error type on the server for authorization errors

**Review by dimapopovych — CHANGES_REQUESTED** [review]

File: frontend/src/bundles/auth/auth-api.ts (hunk 1 above)
Comment (ivanbilobrov07) [reply frontend/src/bundles/auth/auth-api.ts:54]: to set token to null in db, or it is not necessary?

Comment (dimapopovych) [reply frontend/src/bundles/auth/auth-api.ts:54]: but we don't have token in the database currently

Comment (ivanbilobrov07) [reply frontend/src/bundles/auth/auth-api.ts:54]: Didn’t notice it 🤦‍♂️

**Review by nikita-remeslov — APPROVED** [review]

**Pull request merged**